dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -74,7 +82,20 @@ func (cv *CandidateValidation) processMessages(wg *sync.WaitGroup) {
 			case ValidateFromChainState:
 				// TODO: implement functionality to handle ValidateFromChainState, see issue #3919
 			case ValidateFromExhaustive:
-				// TODO: implement functionality to handle ValidateFromExhaustive, see issue #3547
+				result, err := validateFromExhaustive(cv.ValidationHost, msg.PersistedValidationData,
+					msg.ValidationCode, msg.CandidateReceipt, msg.PoV)
+				if err != nil {
+					logger.Errorf("failed to validate from exhaustive: %w", err)
+					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+						Data: *result,
+						Err:  err,
+					}
```

Comment:
If there is validation error, Data should be nil
```suggestion
					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
						Err:  err,
					}
```

Reply:
Updated.